Error opening Lens on Beta 13

**Describe the bug**
Version 5.0.0-beta13 is erroring as soon as it is opened with the following messages:
```
Component stack:

    at HotbarMenu (http://localhost:61296/build/Lens.js:553327:30)
    at div
    at ClusterManager (http://localhost:61296/build/Lens.js:555131:38)
    at Route (http://localhost:61296/build/Lens.js:18628:29)
    at Switch (http://localhost:61296/build/Lens.js:18798:29)
    at ErrorBoundary (http://localhost:61296/build/Lens.js:475027:9)
    at Router (http://localhost:61296/build/Lens.js:18291:30)
    at LensApp (http://localhost:61296/build/Lens.js:555937:24)
    at ThemeProvider (http://localhost:61296/build/Lens.js:555993:24)
```

```
Error stack:

TypeError: Cannot read property 'items' of undefined
    at HotbarMenu.get items [as items] (http://localhost:61296/build/Lens.js:553363:35)
    at HotbarMenu.renderGrid (http://localhost:61296/build/Lens.js:553377:21)
    at HotbarMenu.render (http://localhost:61296/build/Lens.js:553401:111)
    at allowStateChanges (http://localhost:61296/build/Lens.js:1329:12)
    at http://localhost:61296/build/Lens.js:471708:59
    at trackDerivedFunction (http://localhost:61296/build/Lens.js:1881:18)
    at Reaction.track (http://localhost:61296/build/Lens.js:2409:18)
    at HotbarMenu.reactiveRender [as render] (http://localhost:61296/build/Lens.js:471706:14)
    at HotbarMenu.makeComponentReactive (http://localhost:61296/build/Lens.js:471721:25)
    at HotbarMenu.target.render (http://localhost:61296/build/Lens.js:471634:34)
    at qi (http://localhost:61296/build/Lens.js:442364:188)
    at pi (http://localhost:61296/build/Lens.js:442363:173)
    at ck (http://localhost:61296/build/Lens.js:442446:427)
    at bk (http://localhost:61296/build/Lens.js:442427:347)
    at ak (http://localhost:61296/build/Lens.js:442427:278)
    at Tj (http://localhost:61296/build/Lens.js:442427:138)
```


**To Reproduce**
Steps to reproduce the behavior:
1. Open the application on the given version
2. Error appears when window is opened
3. See error

**Expected behavior**
Application opens normally

**Screenshots**
<img width="1438" alt="Screenshot 2021-06-29 at 15 06 40" src="https://user-images.githubusercontent.com/1015506/123812479-086a5580-d8ec-11eb-848f-7f281d41f038.png">


**Environment (please complete the following information):**
- Lens Version: 5.0.0-beta.13
- OS: MacOS 10.14.6
- Installation method (e.g. snap or AppImage in Linux): DMG by dragging and dropping into the Applications folder

**Logs:**
```
 ~  /Applications/Lens.app/Contents/MacOS/Lens
info: 📟 Setting Lens as protocol client for lens://
info: 📟 Protocol client register succeeded ✅
info: 🚀 Starting Lens from "/Users/my-user/Library/Application Support/Lens"
info: 🐚 Syncing shell environment
info: 💾 Loading stores
info: [STORE]: LOADED from /Users/my-user/Library/Application Support/Lens/lens-user-store.json
info: [STORE]: LOADED from /Users/my-user/Library/Application Support/Lens/lens-cluster-store.json
info: [STORE]: LOADED from /Users/my-user/Library/Application Support/Lens/lens-hotbar-store.json
[mobx] Out of bounds read: 0
info: [STORE]: LOADED from /Users/my-user/Library/Application Support/Lens/lens-extensions.json
info: [STORE]: LOADED from /Users/my-user/Library/Application Support/Lens/lens-filesystem-provisioner-store.json
info: [STORE]: LOADED from /Users/my-user/Library/Application Support/Lens/lens-weblink-store.json
info: 🔌 Starting LensProxy
info: [LENS-PROXY]: Proxy server has started at 127.0.0.1:64581
info: 🔎 Testing LensProxy connection ...
(node:87124) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'id' of undefined
    at da.fromStore (/Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:139250)
    at executeAction (/Applications/Lens.app/Contents/Resources/app.asar/node_modules/mobx/dist/mobx.cjs.development.js:1130:15)
    at da.fromStore (/Applications/Lens.app/Contents/Resources/app.asar/node_modules/mobx/dist/mobx.cjs.development.js:1114:12)
    at da.load (/Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:31646)
    at new da (/Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:138742)
    at Function.createInstance (/Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:28770)
    at App.<anonymous> (/Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:235354)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
(node:87124) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:87124) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
info: ⚡ LensProxy connection OK
info: 🖥️  Starting WindowManager
info: 🧩 Initializing extensions
info: [EXTENSION-DISCOVERY] loading extensions from /Users/my-user/Library/Application Support/Lens
info: [EXTENSION-INSTALLER] installing dependencies at /Users/my-user/Library/Application Support/Lens
info: [WINDOW-MANAGER]: Loading Main window from url: http://localhost:64581 ...
info: [EXTENSION-INSTALLER] dependencies installed at /Users/my-user/Library/Application Support/Lens
info: [EXTENSION-DISCOVERY] watching extension add/remove in /Users/my-user/.k8slens/extensions
Native extraction module not loaded, msgpackr will still run, but with decreased performance. Cannot find module 'msgpackr-extract'
Lens Cloud Extension activated
info: [STORE]: LOADED from /Users/my-user/Library/Application Support/Lens/extension-store/lenscloud-lens-extension/auth-token-store.json
error: [EXTENSIONS-LOADER]: can't load main for "lens-license": Error: Cannot find module '/Users/my-user/Library/Application Support/Lens/node_modules/lens-license/dist/main.js'
Require stack:
- /Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js
-  {"extension":{"id":"/Users/my-user/Library/Application Support/Lens/node_modules/lens-license/package.json","absolutePath":"/Applications/Lens.app/Contents/Resources/extensions/license-menu-item/lens-license-5.0.0-beta.13.20210628.2.tgz","manifestPath":"/Users/my-user/Library/Application Support/Lens/node_modules/lens-license/package.json","manifest":{"name":"lens-license","version":"5.0.0-beta.13.20210628.2","description":"Lens IDE license menu item","main":"dist/main.js","scripts":{"build":"webpack -p & npm pack","dev":"webpack --watch","test":"jest --passWithNoTests --env=jsdom src $@","clean":"rm -rf *.tgz dist/"},"files":["dist/**/*"],"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","@types/node":"^14.14.14","@types/react":"^17.0.0","@types/webpack":"^4.41.17","conf":"^7.1.2","jest":"^26.6.3","mobx":"^6.3.2","react":"^16.13.1","ts-loader":"^8.0.4","ts-node":"^9.0.0","typescript":"^4.0.3","webpack":"^4.44.2","webpack-cli":"^3.3.11"}},"isBundled":true,"isEnabled":true,"isCompatible":true}}
error: [EXTENSIONS-LOADER]: can't load main for "lens-telemetry": TypeError: Cannot create a second singleton while creating a first {"extension":{"id":"/Users/my-user/Library/Application Support/Lens/node_modules/lens-telemetry/package.json","absolutePath":"/Applications/Lens.app/Contents/Resources/extensions/telemetry/lens-telemetry-5.0.0-beta.13.20210628.2.tgz","manifestPath":"/Users/my-user/Library/Application Support/Lens/node_modules/lens-telemetry/package.json","manifest":{"name":"lens-telemetry","version":"5.0.0-beta.13.20210628.2","description":"Lens IDE telemetry","main":"dist/main.js","renderer":"dist/renderer.js","lens":{"metadata":{},"styles":[]},"scripts":{"build":"webpack -p && npm pack","dev":"webpack --watch","test":"jest --passWithNoTests --env=jsdom src $@"},"files":["dist/**/*"],"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","@types/analytics-node":"^3.1.3","@types/node":"^14.14.14","@types/react":"^17.0.0","@types/universal-analytics":"^0.4.4","analytics-node":"^3.4.0-beta.3","conf":"^7.1.2","jest":"^26.6.3","mobx":"^6.3.2","mobx-react":"^7.1.0","node-machine-id":"^1.1.12","react":"^16.13.1","ts-loader":"^8.0.4","typescript":"^4.0.3","universal-analytics":"^0.4.23","webpack":"^4.44.2","webpack-cli":"^3.3.11"}},"isBundled":true,"isEnabled":true,"isCompatible":true}}
error: [EXTENSION]: failed to activate lens-survey@5.0.0-beta.13.20210628.2: Error: [MobX] No annotations were passed to makeObservable, but no decorated members have been found either
[CLUSTER-CONNECT] cluster proxy started on port 64603
[CLUSTER-CONNECT] cluster proxy is using aes-256-gcm for end-to-end encryption
info: [EXTENSION]: enabled lenscloud-lens-extension@5.0.0-beta.13.20210628.2
info: [KUBECONFIG-SYNC]: starting requested syncs
info: 📡 Checking for app updates
info: Checking for update
info: [KUBECONFIG-SYNC]: starting sync of file/folder {"filePath":"/Users/my-user/.kube"}
warn: [KUBECONFIG-SYNC]: failed to start watching changes: Error: ENOENT: no such file or directory, stat '/Users/my-user/Library/Application Support/Lens/kubeconfigs'
info: [WINDOW-MANAGER]: Main window loaded
info: [WINDOW-MANAGER]: Main window has reported that it has loaded
info: Update for version 5.0.0-beta.13.20210628.2 is not available (latest version: 5.0.0-beta.13.20210628.2, downgrade is disallowed).
```

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0t...tCg==
    server: https://192.168.xxx.xxx:8443
  name: demo
- cluster:
    certificate-authority-data: LS0tLS...LQo=
    server: https://34.xx.212.xxx
  name: gke_circle-dev-228708_europe-west6_circle-dev-k8s-cluster
- cluster:
    certificate-authority-data: LS0t...tLQo=
    server: https://34.xx.182.xxx
  name: gke_circle-prod-241909_europe-west6_circle-prod-k8s-cluster
- cluster:
    certificate-authority-data: LS0t...S0K
    server: https://34.xx.53.xx
  name: gke_circle-test-233809_europe-west6_circle-test-k8s-cluster
contexts:
- context:
    cluster: demo
    namespace: demo
    user: demo-user
  name: demo
- context:
    cluster: gke_circle-dev-jjjjjj_europe-west6_circle-dev-k8s-cluster
    user: gke_circle-dev-jjjjj_europe-west6_circle-dev-k8s-cluster
  name: gke_circle-dev-jjjjj_europe-west6_circle-dev-k8s-cluster
- context:
    cluster: gke_circle-prod-xxxxxx_europe-west6_circle-prod-k8s-cluster
    user: gke_circle-prod-xxxxxx_europe-west6_circle-prod-k8s-cluster
  name: gke_circle-prod-xxxxxx_europe-west6_circle-prod-k8s-cluster
- context:
    cluster: gke_circle-test-yyyyyy_europe-west6_circle-test-k8s-cluster
    user: gke_circle-test-yyyyyy_europe-west6_circle-test-k8s-cluster
  name: gke_circle-test-yyyyyy_europe-west6_circle-test-k8s-cluster
current-context: gke_circle-dev-yyyyyy_europe-west6_circle-dev-k8s-cluster
kind: Config
preferences: {}
users:
- name: demo-user
  user:
    client-key-data: LS0...Cg==
    token: eyJ...xw
- name: gke_circle-dev-yyyyyy_europe-west6_circle-dev-k8s-cluster
  user:
    auth-provider:
      config:
        access-token: ya29...Gs
        cmd-args: config config-helper --format=json
        cmd-path: /Users/my-user/google-cloud-sdk/bin/gcloud
        expiry: "2021-06-29T11:00:54Z"
        expiry-key: '{.credential.token_expiry}'
        token-key: '{.credential.access_token}'
      name: gcp
- name: gke_circle-prod-xxxxxx_europe-west6_circle-prod-k8s-cluster
  user:
    auth-provider:
      config:
        access-token: ya29...jQ
        cmd-args: config config-helper --format=json
        cmd-path: /Users/my-user/google-cloud-sdk/bin/gcloud
        expiry: "2021-05-18T12:52:07Z"
        expiry-key: '{.credential.token_expiry}'
        token-key: '{.credential.access_token}'
      name: gcp
- name: gke_circle-test-jjjjjj_europe-west6_circle-test-k8s-cluster
  user:
    auth-provider:
      config:
        access-token: ya29...Sw
        cmd-args: config config-helper --format=json
        cmd-path: /Users/my-user/google-cloud-sdk/bin/gcloud
        expiry: "2021-05-31T12:06:26Z"
        expiry-key: '{.credential.token_expiry}'
        token-key: '{.credential.access_token}'
      name: gcp
```

**Additional context**
Add any other context about the problem here.

--- a/src/common/hotbar-store.ts
+++ b/src/common/hotbar-store.ts
@@ -26,6 +26,7 @@ import * as uuid from "uuid";
 import isNull from "lodash/isNull";
 import { toJS } from "./utils";
 import { CatalogEntity } from "./catalog";
+import { catalogEntity } from "../main/catalog-sources/general";
 
 export interface HotbarItem {
   entity: {
@@ -94,7 +95,15 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
 
   @action
   protected async fromStore(data: Partial<HotbarStoreModel> = {}) {
-    this.hotbars = data.hotbars;
+    if (!data.hotbars || !data.hotbars.length) {
+      this.hotbars = [{
+        id: uuid.v4(),
+        name: "Default",
+        items: this.defaultHotbarInitialItems,
+      }];
+    } else {
+      this.hotbars = data.hotbars;
+    }
 
     if (data.activeHotbarId) {
       if (this.getById(data.activeHotbarId)) {
@@ -105,6 +114,16 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
     if (!this.activeHotbarId) {
       this.activeHotbarId = this.hotbars[0].id;
     }
+  }
+
+  get defaultHotbarInitialItems() {
+    const { metadata: { uid, name, source } } = catalogEntity;
+    const initialItem = { entity: { uid, name, source }};
+
+    return [
+      initialItem,
+      ...Array.from(Array(defaultHotbarCells - 1).fill(null))
+    ];
   }
 
   getActive() {

